pycbc/waveform/waveform.py

```diff
@@ -88,6 +88,15 @@ def _lalsim_fd_waveform(**p):
     lalsimulation.SimInspiralSetSpinOrder(flags, p['spin_order'])
     lalsimulation.SimInspiralSetTidalOrder(flags, p['tidal_order'])
 
+    # SpinTaylorF2 is only single spin. Fall back to TaylorF2 if aligned-spin
+    # and spin2z is not zero. If precessing and spin2 non-zero the code will
+    # fail later.
+    if p['approximant'] == 'SpinTaylorF2' and (not p['spin2z'] == 0):
+        if p['spin2x'] == 0 and p['spin2y'] == 0 and p['spin1x'] == 0 and p['spin1y'] == 0:
+            # Fall back to standard F2
+            p['approximant'] = 'TaylorF2'
+
+
```

Comment:
Can you move this to inside get_two_pol_waveform_filter? I'm a bit wary of allowing this to silently happen even when calling get_fd_waveform, which should just be a thin wrapper around the underlying waveform generators.